pkg/adafruitio/adafruitio.go

```diff
@@ -78,6 +81,74 @@ func (c *Client) Record(feed, value string, timestamp time.Time) error {
 	return nil
 }
 
+// Feeds returns all feeds of an Adafruit user.
+func (c *Client) Feeds() ([]Feed, error) {
+	// Create request.
+	req, err := http.NewRequest(
+		http.MethodGet,
+		"https://io.adafruit.com/api/v2/"+c.username+"/feeds",
+		nil,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not set API request headers")
+	}
+
+	// Send request.
+	res, err := c.send(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not send API request")
+	}
+	defer res.Body.Close()
+
+	// Marshal response into Feed struct.
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read response body")
+	}
```

Comment:
All the request code up to here seems semantically identical. It's probably better to refactor these common semantics into a single method.